kits/apptentive-kit/src/main/java/com/mparticle/kits/ApptentiveKit.java

```diff
@@ -0,0 +1,250 @@
+	private static final String API_KEY = "appKey";
+	private static final String CUSTOM_DIMENSIONS = "customDimensions";
+	private static final String RAW_LTV = "trackClvAsRawValue";
+	private JSONArray customDimensionJson = null;
+	private boolean trackAsRawLtv = false;
+	private ApptentiveActivityLifecycleCallbacks callbacks;
+
+	@Override
+	protected AbstractKit update() {
+		if (callbacks == null) {
+			callbacks = new ApptentiveActivityLifecycleCallbacks(context.getApplicationContext());
+		}
+		/* Note mParticle will delegate lifecycle management to the above callbacks. No need to register ApptentiveActivityLifecycleCallbacks
+         * through Apptentive. But do need to initialize Apptentive
+         *
+         */
+		ApptentiveInternal.createInstance(context.getApplicationContext(), properties.get(API_KEY));
+
+		return this;
+
+	}
+
+	@Override
+	public Object getInstance(Activity activity) {
+		return null;
+	}
+
+	@Override
+	public String getName() {
+		return "Apptentive";
+	}
+
+	@Override
+	public boolean isOriginator(String uri) {
+		return uri != null && uri.toLowerCase().contains("Apptentive.com");
+	}
+
+
+	@Override
+	public void setUserIdentity(String id, MParticle.IdentityType identityType) {
+		if (identityType.equals(MParticle.IdentityType.Email)) {
+			Apptentive.setPersonEmail(id);
+		} else if (identityType.equals(MParticle.IdentityType.CustomerId)) {
+			Apptentive.setPersonName(id);
+		}
+	}
+
+	@Override
+	void setUserAttributes(JSONObject mUserAttributes) {
+		HashSet<String> attributeKeys = null;
+		if (customDimensionJson != null) {
+			try {
+				for (int i = 0; i < customDimensionJson.length(); i++) {
+					JSONObject dimension = customDimensionJson.getJSONObject(i);
+					if (dimension.getString("maptype").equals("UserAttributeClass.Name")) {
+						String attributeName = dimension.getString("map");
+						String attributeValue = mUserAttributes.optString(attributeName);
+						if (!TextUtils.isEmpty(attributeValue)) {
+							if (attributeKeys == null) {
+								attributeKeys = new HashSet<String>();
+							}
+							attributeKeys.add(attributeName);
+							Apptentive.addCustomPersonData(attributeName, attributeValue);
+						}
+					}
+				}
+			} catch (Exception e) {
+				ConfigManager.log(MParticle.LogLevel.DEBUG, "Exception while mapping mParticle user attributes to Apptentive custom dimensions: " + e.toString());
+			}
+		}
+		Iterator<String> keys = mUserAttributes.keys();
+		String firstName = "";
+		String lastName = "";
+		while (keys.hasNext()) {
+			String key = keys.next();
+			if (attributeKeys == null || !attributeKeys.contains(key)) {
+				String value = mUserAttributes.optString(key);
+				if (key.equalsIgnoreCase(MParticle.UserAttributes.FIRSTNAME)) {
+					firstName = value;
+				} else if (key.equalsIgnoreCase(MParticle.UserAttributes.LASTNAME)) {
+					lastName = value;
+				}
+			}
+		}
+		Apptentive.setPersonName(firstName + lastName);
```

Comment:
You may want to consider formatting this with a space in between firstName and lastName in the case where both are populated.